roxie/roxiemem/roxiemem.cpp

```diff
@@ -628,12 +637,12 @@ static void *suballoc_aligned(size32_t pages, bool returnNullWhenExhausted)
         unsigned i;
         for (i = heapLWM; i < heapBitmapSize; i++)
         {
-            unsigned hbi = heapBitmap[i];
+            heap_t hbi = heapBitmap[i];
             if (hbi)
             {
                 const unsigned pos = countTrailingUnsetBits(hbi);
-                const unsigned mask = 1U << pos;
-                const unsigned match = i*UNSIGNED_BITS + pos;
+                const heap_t mask = ((heap_t)1U) << pos;
```

Comment:
I coded like that so it wasn't conditional- and we used to require I64C() macro for 64bit constants - may not be required now we are using c++11.